packages/manager/apps/dedicated/client/app/ip/ip/firewall/ip-ip-firewall.controller.js

```diff
@@ -40,69 +42,77 @@ angular
+                  options = [];
+                  if (result.fragments) {
+                    options.push(
+                      $translate.instant('ip_firewall_rule_fragments'),
+                    );
+                  }
+                  if (result.tcpOption) {
+                    options.push(result.tcpOption);
+                  }
 
-                if (
-                  $scope.rules &&
-                  $scope.rules.list &&
-                  $scope.rules.list.results &&
-                  $scope.rules.list.results.length
-                ) {
-                  angular.forEach($scope.rules.list.results, (result, i) => {
-                    options = [];
-                    if (result.fragments) {
-                      options.push(
-                        $translate.instant('ip_firewall_rule_fragments'),
-                      );
-                    }
-                    if (result.tcpOption) {
-                      options.push(result.tcpOption);
-                    }
+                  $scope.rules.list.results[i].options = options.join('<br/>');
 
-                    $scope.rules.list.results[i].options = options.join(
-                      '<br/>',
-                    );
+                  // Go poll
+                  if (
+                    result.state === 'CREATION_PENDING' ||
+                    result.state === 'REMOVAL_PENDING'
+                  ) {
+                    IpFirewall.pollFirewallRule(
+                      $scope.selectedBlock,
+                      $scope.selectedIp,
+                      result.sequence,
+                    ).then(() => {
+                      reloadRules();
+                    });
+                  }
+                });
+              }
+            },
+            (reason) => {
+              $scope.rulesLoadingError = reason.message;
+            },
+          )
+          .finally(() => {
+            $scope.rulesLoading = false;
+          });
+      }
 
-                    // Go poll
-                    if (
-                      result.state === 'CREATION_PENDING' ||
-                      result.state === 'REMOVAL_PENDING'
-                    ) {
-                      IpFirewall.pollFirewallRule(
-                        $scope.selectedBlock,
-                        $scope.selectedIp,
-                        result.sequence,
-                      ).then(() => {
-                        reloadRules();
-                      });
-                    }
-                  });
-                }
-              },
-              (reason) => {
-                $scope.rulesLoadingError = reason.message;
-              },
-            )
-            .finally(() => {
-              $scope.rulesLoading = false;
-            });
+      $scope.$on('ips.firewall.informations.reload', () => {
+        reloadRules();
+      });
+
+      $scope.loadRules = function loadRules(rulesCount, offset) {
+        if (!$scope.selectedIp && !$scope.selectedBlock) {
```

Comment:
Is `$scope.selectedBlock` test useful here ? You could set `$scope.selectedBlock = data.ip;` in all cases no ?

Reply:
Thanks for the review. No, I can not set it to data.ip in all cases. There are two cases.
First: user comes from the action menu. In this case, both selectedIp and selectedIpBlock are provided.
Second: User enters URL directly. In this case, only IP is available, need to fetch the ipBlock based on the IP.